### PR TITLE
`ReadList[]`  handling `Null` and `Hold[]` expressions

### PR DESCRIPTION
**Pull request by rocky**
See the `ReadList[]` example in https://github.com/Mathics3/Mathics3-Rubi/pull/2

### CODE REVIEW

**Comment by rocky** [comment]
@aravindh-krishnamoorthy The ReadList[] example had a nice reproducible example.  Others in that list like resetting RecursionLimit seem pretty simple too, but a small reproducible example would be nice to have.

**Comment by aravindh-krishnamoorthy** [comment]
> @aravindh-krishnamoorthy The ReadList[] example had a nice reproducible example. Others in that list like resetting RecursionLimit seem pretty simple too, but a small reproducible example would be nice to have.

Indeed! I will work on them now!

File: test/builtin/files_io/test_files.py [hunk 1]
```diff
@@ -295,6 +295,12 @@ def test_close():
             "{{a, 1}}",
             "",
         ),
+        (
+            'ReadList[StringToStream["(**)"], Expression]',
+            None,
+            "{Null}",
+            "",
+        ),
```
Comment (aravindh-krishnamoorthy) [inline test/builtin/files_io/test_files.py:303]:
```suggestion
        ),
        (
            'ReadList[StringToStream["Hold[1+2]"], Expression]',
            None,
            "{Hold[1 + 2]}",
            "",
        ),
```

**Comment by aravindh-krishnamoorthy** [comment]
Perhaps good to mention `Hold` as well in the PR title?

**Comment by aravindh-krishnamoorthy** [comment]
LGTM!

**Pull request merged**